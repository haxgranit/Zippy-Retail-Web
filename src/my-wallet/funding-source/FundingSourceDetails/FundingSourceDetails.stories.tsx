import { ComponentStory, ComponentMeta } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import FundingSourceDetails from './FundingSourceDetails';

export default {
  title: 'my-wallet/funding-source/FundingSourceDetails',
  component: FundingSourceDetails,
} as ComponentMeta<typeof FundingSourceDetails>;

const Template: ComponentStory<typeof FundingSourceDetails> = () => (
  <BrowserRouter>
    <FundingSourceDetails />
  </BrowserRouter>
);

export const Default = Template.bind({});
Default.args = {};
Default.parameters = {};
