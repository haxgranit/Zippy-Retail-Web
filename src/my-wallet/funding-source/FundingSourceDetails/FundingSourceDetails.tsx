import { useMsal } from '@azure/msal-react';
import { useState } from 'react';
import { Button, Form } from 'react-bootstrap';
import { useNavigate } from 'react-router-dom';
import Api from '../../../api';
import PageContainer from '../../../common/PageContainer';

export default function FundingSourceDetails() {
  const [bankName, setBankName] = useState<string>('');
  const [accountNumber, setAccountNumber] = useState<string>('');
  const [institutionNumber, setInstitutionNumber] = useState<string>('');
  const [transitNumber, setTransitNumber] = useState<string>('');
  const [isDefault, setIsDefault] = useState<boolean>(false);
  const [isLoading, setIsLoading] = useState<boolean>(false);

  const { instance, accounts } = useMsal();

  const navigate = useNavigate();

  function saveAccount() {
    setIsLoading(true);
    new Api(instance, accounts[0])
      .postFundingSource({
        displayName: bankName,
        isDefault,
        bankAccount: {
          accountNumber,
          institutionNumber: +institutionNumber,
          transitNumber: +transitNumber,
        },
      })
      .then(() => {
        setIsLoading(false);
        navigate('/my-wallet/funding-source');
      })
      .catch((error) => {
        console.log('error', error);
        setIsLoading(false);
      });
  }
  return (
    <PageContainer title="Your Wallet" subTitle="Made Fun With Zippy!" backdropImage="background-funding-image">
      <div className="fundingTitle">
        Funding Source
      </div>

      <div className="bankForm">
        <form action="">
          <Form.Control
            type="text"
            placeholder="Nickname"
            value={bankName}
            onChange={(evt) => setBankName(evt.target.value)}
          />
          <Form.Control
            type="number"
            placeholder="Account Number"
            value={accountNumber}
            onChange={(evt) => setAccountNumber(evt.target.value)}
          />
          <Form.Control
            type="number"
            placeholder="Institution Number"
            value={institutionNumber}
            onChange={(evt) => setInstitutionNumber(evt.target.value)}
          />
          <Form.Control
            type="number"
            placeholder="Transit Number"
            value={transitNumber}
            onChange={(evt) => setTransitNumber(evt.target.value)}
          />
          <div style={{ display: 'flex', justifyContent: 'flex-start', alignItems: 'center' }}>
            <Form.Check
              type="radio"
              className="actions"
              label="Mark this Account as a default"
              name="actions2"
              onChange={(evt) => { setIsDefault(evt.target.checked); }}
            />
          </div>
        </form>

      </div>

      <div className="action">
        <Button
          disabled={isLoading}
          className="zippy-btn"
          onClick={() => {
            saveAccount();
          }}
        >
          {isLoading && <div className="loading spinner-border" role="status" />}
          Save
        </Button>
      </div>
    </PageContainer>
  );
}
