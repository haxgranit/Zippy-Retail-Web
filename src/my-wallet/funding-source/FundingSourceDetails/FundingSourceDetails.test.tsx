import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import FundingSourceDetails from './FundingSourceDetails';

const ReactTestRenderer = require('react-test-renderer');

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe('FundingSourceList Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <BrowserRouter>
        <FundingSourceDetails />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
