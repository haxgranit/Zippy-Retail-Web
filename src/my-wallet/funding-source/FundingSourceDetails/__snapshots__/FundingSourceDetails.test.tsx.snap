// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FundingSourceList Component matches the snapshot 1`] = `
<div
  className=""
>
  <div
    className="zippy-cash-container-wrap"
  >
    <div
      className="backdrop background-funding-image"
    />
    <div
      className="zippy-cash-container"
    >
      <div
        className="zippy-cash-container-header"
      >
        <h1
          className="title"
        >
          Your Wallet
        </h1>
        <h2
          className="catch"
        >
          Made Fun With Zippy!
        </h2>
      </div>
      <div
        className="zippy-cash-container-body"
      >
        <div
          className="fundingTitle"
        >
          Funding Source
           
        </div>
        <div
          className="bank-form"
        >
          <form
            action=""
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Bank Name"
              type="text"
              value=""
            />
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Account Name"
              type="text"
              value=""
            />
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Account Number"
              type="number"
              value=""
            />
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Institution Number"
              type="number"
              value=""
            />
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Transit Number"
              type="number"
              value=""
            />
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "flex-start",
                }
              }
            >
              <div
                className="actions form-check"
              >
                <input
                  className="form-check-input"
                  disabled={false}
                  id="interac-e-transfer"
                  name="actions2"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  className="form-check-label"
                  htmlFor="interac-e-transfer"
                  title=""
                >
                  Mark this Account as a default
                </label>
              </div>
            </div>
          </form>
        </div>
        <div
          className="action"
        >
          <button
            className="zippy-btn btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
